bertopic/_bertopic.py

```diff
@@ -467,7 +469,6 @@ def fit_transform(
             # All documents matches zero-shot topics
             documents = assigned_documents
             embeddings = assigned_embeddings
-        topics_before_reduction = self.topics_
 
         # Sort and Map Topic IDs by their frequency
         if not self.nr_topics:
```

Comment:
Why does sorting mappings *not* occur if `nr_topics` is passed?

Reply:
The `topics_before_reduction` were used for calculating the probabilities and after I made a couple of changes, it started giving index errors when attempting to access the similarity matrix.

Reply:
Specifically, if I add `nr_topics="auto"` when training BERTopic using the latest commit in this PR, I get the following error:

```python
from bertopic.representation import KeyBERTInspired

# KeyBERT
keybert_model = KeyBERTInspired()

# Pass the above models to be used in BERTopic
topic_model = UpdatedBERTopic(
    embedding_model=embedding_model,
    umap_model=umap_model,
    hdbscan_model=hdbscan_model,
    verbose=True,
    zeroshot_topic_list=zeroshot_topic_list,
    zeroshot_min_similarity=.5,
    nr_topics="auto",
    calculate_probabilities=True,
    representation_model=keybert_model
)
topic_model = topic_model.fit(abstracts, embeddings)
```

Then that gives me the following error:

```bash
IndexError                                Traceback (most recent call last)
Cell In[13], line 18
      6 # Pass the above models to be used in BERTopic
      7 topic_model = UpdatedBERTopic(
      8     embedding_model=embedding_model,
      9     umap_model=umap_model,
   (...)
     16     representation_model=keybert_model
     17 )
---> 18 topic_model = topic_model.fit(abstracts, embeddings)

File [~\Documents\Projects\BERTopic\bertopic\_bertopic.py:364](http://localhost:8888/lab/tree/Documents/Projects/~/Documents/Projects/BERTopic/bertopic/_bertopic.py#line=363), in BERTopic.fit(self, documents, embeddings, images, y)
    322 def fit(
    323     self,
    324     documents: List[str],
   (...)
    327     y: Union[List[int], np.ndarray] = None,
    328 ):
    329     """Fit the models (Bert, UMAP, and, HDBSCAN) on a collection of documents and generate topics.
    330 
    331     Arguments:
   (...)
    362     ```
    363     """
--> 364     self.fit_transform(documents=documents, embeddings=embeddings, y=y, images=images)
    365     return self

Cell In[12], line 300, in UpdatedBERTopic.fit_transform(self, documents, embeddings, images, y)
    292     else:
    293         # Use `topics_before_reduction` because `self.topics_` may have already been updated from
    294         # reducing topics, and the original probabilities are needed for `self._map_probabilities()`
    295         probabilities = sim_matrix[
    296             np.arange(len(documents)),
    297             np.array(topics_before_reduction) + self._outliers,
    298         ]
--> 300 self.probabilities_ = self._map_probabilities(probabilities, original_topics=True)
    301 predictions = documents.Topic.to_list()
    303 return predictions, self.probabilities_

File [~\Documents\Projects\BERTopic\bertopic\_bertopic.py:4581](http://localhost:8888/lab/tree/Documents/Projects/~/Documents/Projects/BERTopic/bertopic/_bertopic.py#line=4580), in _map_probabilities(self, probabilities, original_topics)
   4578             if to_topic != -1 and from_topic != -1:
   4579                 mapped_probabilities[:, to_topic] += probabilities[:, from_topic]
-> 4581         return mapped_probabilities
   4583 return probabilities

IndexError: index 71 is out of bounds for axis 1 with size 71

```